File: java/matrix/src/main/java/Matrix.java
import java.util.Arrays;

class Matrix {

    int[][] matrix;

    Matrix(String matrixAsString) {
        matrix = Arrays.stream(matrixAsString.split("\n"))
                .map(str -> Arrays.stream(str.split(" ")).mapToInt(Integer::parseInt).toArray())
                .toArray(int[][]::new);
    }

    int[] getRow(int rowNumber) {
        return matrix[rowNumber - 1];
    }

    int[] getColumn(int columnNumber) {
        return Arrays.stream(matrix).mapToInt(arr -> arr[columnNumber - 1]).toArray();
    }
}

File: kotlin/matrix/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {

    private val matrix: List<List<Int>> = matrixAsString
        .split("\n")
        .map {
            it.split(" ")
                .map(String::toInt)
        }

    fun column(colNr: Int): List<Int> {
        return matrix.map { it[colNr-1] }
    }

    fun row(rowNr: Int): List<Int> {
        return matrix[rowNr - 1]
    }
}
